crates/matrix-sdk/src/encryption/recovery/mod.rs

```diff
@@ -285,11 +297,36 @@ impl Recovery {
     #[instrument(skip_all)]
     pub async fn disable(&self) -> Result<()> {
         self.client.encryption().backups().disable().await?;
+
         // Why oh why, can't we delete account data events?
+        //
+        // Alright, let's attempt to "delete" the content of our current default key,
+        // for this we first need to check if there is a default key, then
+        // deserialize the content and find out the key ID.
+        //
+        // Then we finally set the event to an empty JSON content.
+        if let Ok(Some(default_event)) =
```

Comment:
As an aside, I think that referring to account data as an "event" which can be "deleted" or changed in any way, is a confusing anti-pattern. One of the defining properties of an "event" in most environments is that it's something that happened in the past, so changing it is nonsense.

(I know that the spec uses that term, but I'd like that to change (https://github.com/matrix-org/matrix-spec/issues/2030), and we can avoid spreading the mess any further in the meantime.)

Reply:
So, what should we call them?

Reply:
"Account data". Or possibly, "Account data entries".